### PR TITLE
add AccessQueryRunner to cadc-repo-server

### PR DESCRIPTION
**Pull request by pdowler**
uses ReadAccessDAO.getArtifactAccess utility query to return an ArtifactAccess object

### CODE REVIEW

File: caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/AccessQueryRunner.java [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+                    URI guri = URI.create(gms.toASCIIString() + "?" + s);
+                    metaReadAccessGroups.add(guri);
+                }
+                for (String s : raa.dataReadAccessGroups) {
+                    URI guri = URI.create(gms.toASCIIString() + "?" + s);
+                    dataReadAccessGroups.add(guri);
+                }
+                ArtifactAccess aa = AccessUtil.getArtifactAccess(raa.artifact,
+                        raa.metaRelease, metaReadAccessGroups, raa.dataRelease, dataReadAccessGroups);
+                log.warn("raw: " + raa.releaseType
```
Comment (brianmajor) [inline caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/AccessQueryRunner.java]:
```suggestion
                log.debug("raw: " + raa.releaseType 
```

File: caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/AccessQueryRunner.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+            throw new RuntimeException("CONFIG: failed to read config from " + config.getAbsolutePath());
+        }
+    }
+    
+    private boolean archiveMatch(URI uri, String pattern) {
+        if (pattern == null) {
+            return false;
+        }
+        if (uri.toASCIIString().contains(pattern)) {
+            log.warn(uri + " vs " + pattern + ": true");
```
Comment (brianmajor) [inline caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/AccessQueryRunner.java]:
```suggestion
            log.debug(uri + " vs " + pattern + ": true");
```

File: caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/AccessQueryRunner.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+            return true;
+        }
+        log.warn(uri + " vs " + pattern + ": false");
```
Comment (brianmajor) [inline caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/AccessQueryRunner.java]:
```suggestion
        log.debug(uri + " vs " + pattern + ": false");
```

Comment (pdowler) [reply caom2-repo-server/src/main/java/ca/nrc/cadc/caom2/repo/AccessQueryRunner.java]: No longer useful so I just removed the log messages

**Pull request merged**